components/credentials/src/main/java/org/cloudfoundry/credhub/generators/CertificateGenerator.java

```diff
@@ -48,10 +55,19 @@ public CertificateCredentialValue generateCredential(final GenerationParameters
       throw new RuntimeException(e);
     }
 
+    final int originalDuration = params.getDuration();
+    Boolean durationOverridden = null;
+
+    params.setDuration(getCertificateDuration(params));
+
+    if (originalDuration != params.getDuration()) {
+      durationOverridden = true;
+    }
+
     if (params.isSelfSigned()) {
       try {
         final String cert = CertificateFormatter.pemOf(signedCertificateGenerator.getSelfSigned(keyPair, params));
-        return new CertificateCredentialValue(cert, cert, privatePem, null, params.isCa(), params.isSelfSigned(), true, false);
+        return new CertificateCredentialValue(cert, cert, privatePem, null, null, params.isCa(), params.isSelfSigned(), true, false, durationOverridden);
```

Comment:
It appears that you create a variable of type `Boolean`, but you never use the value `false`.

If I read this correctly, you initialize it to `null`, optionally set it to `true` on line 64, but no code ever sets it to `false`.

Am I reading this correctly? 
If yes, then maybe `Boolean` isn't the type you want to use.

Reply:
We just pushed a change to the behavior here which I described in the other comment.
1. When the feature is not configured, the field is omitted (null)
2. When the feature is configured but the requested duration was used, the field is false.
3. When the feature is configured and the minimum duration was used instead, the field is true.

Let us know if this makes sense.